### PR TITLE
Bugfix Mission TOAU 15 The Black Coffin

### PR DESCRIPTION
**Pull request by Chronos77**
**_I affirm:_**
- [x] that I agree to Topaz Next's [Limited Contributor License Agreement](https://github.com/topaz-next/topaz/blob/release/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have read the [Contributing Guide](https://github.com/topaz-next/topaz/blob/release/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/topaz-next/topaz/blob/release/CODE_OF_CONDUCT.md)
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

**_Temporary_**:
- [x] that I understand there are large refactoring efforts going on right now, that these efforts touch every single Lua script and binding, and that my pull request might get put on hold to ensure there are not any conflicts with the refactoring work

### CODE REVIEW

File: scripts/globals/instance.lua [hunk 1]
```diff
@@ -1,64 +1,69 @@
...
+        message = 60
+    elseif lastTimeUpdate == 60 and remainingTimeLimit < 30 then
+        message = 30
+    elseif lastTimeUpdate == 30 and remainingTimeLimit < 10 then
+        message = 10
+    end
+
+    if message ~= 0 then
+        for i, player in pairs(players) do
+            if (remainingTimeLimit >= 60) then
```
Comment (ghost) [inline scripts/globals/instance.lua]: Could you please remove the containing parentheses for this conditional to match style guide?

Comment (Chronos77) [reply scripts/globals/instance.lua]: Nice catch ! it's fix!

**Review by EpicTaru — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: scripts/zones/Arrapago_Remnants/instances/arrapago_remnants.lua [hunk 2]
```diff
@@ -2,7 +2,7 @@
 -- Salvage: Arrapago Remnants
 -----------------------------------
 local ID = require("scripts/zones/Arrapago_Remnants/IDs")
-require("scripts/globals/instance")
+local instance_helpers = require ("scripts/globals/instance")
```
Comment (zach2good) [inline scripts/zones/Arrapago_Remnants/instances/arrapago_remnants.lua]: We had a discussion about this in the devs chat; I think we want to only use the Lua module import pattern for more specific globals, like Zone IDs or Zone-specific helpers. For content coming from real globals (scripts/globals) we want to attach the functionality to the tpz object:

```lua
-- instance.lua

tpz = tpz or {}
tpz.instance = {}

tpz.instance.updateInstanceTime(...)
end

---
-- instances/arrapago_remnants.lua 

instance_object.onInstanceTimeUpdate = function(instance, elapsed)
    updateInstanceTime(instance, elapsed, ID.text)
    tpz.instance.updateInstanceTime(instance, elapsed, ID.text)
end
```

**Review by zach2good — CHANGES_REQUESTED** [review]
Thanks for the work! Looking forward to slowly getting TOAU cleaned up 👍 

Watch out for your formatting changes:
```lua
-instance_object.onEventUpdate = function(player, csid, option)
+instance_object.onEventUpdate = function (player, csid, option)
                                   ------^
```

**Comment by Chronos77** [comment]
I made several other modifications to the file format by removing unused required for instance.lua. I will test these modifications and I will remove the WIP when it's ok

**Comment by Chronos77** [comment]
It's works 👍

**Comment by zach2good** [comment]
Content seems fine, it'll be ready to merge once you remove these additional spaces before function arguments:
```lua
entity.onMobSpawn = function (mob)
                        ----^

-- Should be 
entity.onMobSpawn = function(mob)
```

File: scripts/globals/instance.lua [hunk 3]
```diff
@@ -1,64 +1,68 @@
+-----------------------------------
+-- Instance
+-----------------------------------
+tpz = tpz or {}
+tpz.instance = {}
 
-function updateInstanceTime(instance, elapsed, texttable)
-    local players = instance:getChars()
+local function setInstanceLastTimeUpdateMessage (instance, players, remainingTimeLimit)
```
Comment (zach2good) [inline scripts/globals/instance.lua]: Extra space

File: scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua [hunk 4]
```diff
@@ -3,28 +3,26 @@
...
 -----------------------------------
 mixins = {require("scripts/mixins/families/rampart")}
-local ID = require("scripts/zones/Arrapago_Remnants/IDs")
-require("scripts/globals/instance")
 require("scripts/globals/status")
 -----------------------------------
 local entity = {}
 
-entity.onMobSpawn = function(mob)
+entity.onMobSpawn = function (mob)
```
Comment (zach2good) [inline scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua]: Space

File: scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua [hunk 5]
```diff
@@ -3,28 +3,26 @@
...
 end
 
-entity.onMobFight = function(mob, target)
+entity.onMobFight = function (mob, target)
```
Comment (zach2good) [inline scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua]: Space

File: scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua [hunk 6]
```diff
@@ -33,10 +31,10 @@ entity.onMobFight = function(mob, target)
     end
 end
 
-entity.onMobDeath = function(mob, player, isKiller)
+entity.onMobDeath = function (mob, player, isKiller)
```
Comment (zach2good) [inline scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua]: Space

File: scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua [hunk 7]
```diff
@@ -33,10 +31,10 @@ entity.onMobFight = function(mob, target)
...
 end
 
-entity.onMobDespawn = function(mob)
+entity.onMobDespawn = function (mob)
```
Comment (zach2good) [inline scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_1.lua]: Space

File: scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_2.lua [hunk 8]
```diff
@@ -4,35 +4,34 @@
 -----------------------------------
 mixins = {require("scripts/mixins/families/rampart")}
 local ID = require("scripts/zones/Arrapago_Remnants/IDs")
-require("scripts/globals/instance")
 require("scripts/globals/status")
 -----------------------------------
 local entity = {}
 
-entity.onMobSpawn = function(mob)
+entity.onMobSpawn = function (mob)
```
Comment (zach2good) [inline scripts/zones/Arrapago_Remnants/mobs/Archaic_Rampart_2.lua]: Space etc.

**Comment by zach2good** [comment]
No need to keep re-requesting reviews, I check open PRs _at least_ once a day 👍

**Review by zach2good — APPROVED** [review]

**Pull request merged**